docs(metrics): docker container metrics

tdabasinskas opened the pull request:
Fixes #1005.

tdabasinskas committed:
feat(docs): add docker container metrics documentation link

Added new documentation page link in the metrics management section for Docker container metrics to help users monitor their containerized applications

tdabasinskas committed:
feat(docs): add docker metrics dashboard import screenshot

Add visual guide showing the process of importing a dashboard in the Docker metrics documentation

tdabasinskas committed:
docs(metrics-management): add initial Docker container metrics guide

Created a new documentation page that will explain how to monitor Docker container metrics using OpenTelemetry Collector and SigNoz.

This addition will help users understand how to properly set up container monitoring in their environments.

tdabasinskas committed:
docs(docker-container-metrics): add prerequisites section

Added clear prerequisites for monitoring Docker container metrics including:
- Required OTel Collector
- Minimum Docker API version
- OS requirements
- SigNoz deployment options

tdabasinskas committed:
docs(metrics): add instructions for docker container metrics collection

Added documentation explaining how to configure and use the `docker_stats` OpenTelemetry receiver for collecting Docker container metrics. The guide includes:

- Basic receiver configuration with essential metrics
- Instructions for Docker socket mounting when running in container
- Pipeline configuration steps

tdabasinskas committed:
docs(metrics-management): add instructions for sending Docker metrics to SigNoz

Added detailed configuration instructions for setting up exporters in OTel Collector to send Docker metrics to SigNoz. The guide includes:

- Configuration examples for both Cloud and Self-Hosted setups
- Instructions for updating the pipelines section
- Steps to use the correct region and access token for Cloud setup

tdabasinskas committed:
docs(metrics-management): add visualization guide for docker container metrics

Added instructions for visualizing Docker container metrics in SigNoz UI, including:

- Steps to use pre-configured dashboard
- Instructions for importing dashboard JSON
- List of available metric groups (CPU, Memory, Network, Disk)

This helps users better understand how to view and analyze their Docker metrics after collection setup.

tdabasinskas committed:
docs(metrics-management): add conclusion section to docker container metrics guide

Added a conclusion section that summarizes the guide's purpose and provides a link to the official documentation for the complete list of available metrics. This helps users find additional details about metrics they can query in dashboards.

tdabasinskas committed:
feat(content): add docker container metrics documentation

Added new documentation page for Docker container metrics under metrics management section. This helps users understand and monitor Docker container performance metrics effectively.

tdabasinskas committed:
docs(metrics): add docker resource detection processor config

Add documentation for configuring the Docker Resource Detection Processor in OTel Collector to enrich container metrics with additional metadata like hostname and OS information.

This helps users understand how to properly set up metadata enrichment for their Docker container metrics.

tdabasinskas committed:
docs(metrics): add detailed docker metrics visualization guide

Added comprehensive sections explaining different types of metrics visualizations in the Docker container metrics dashboard:

- CPU usage percentage visualization
- Memory usage and limits panels
- Network I/O metrics including bytes received/sent
- Network packets dropped metrics
- Block I/O operations visualization

Each section includes descriptive text and accompanying screenshots to help users better understand the metrics being displayed. Also added the `data-zoomable` attribute to images for better viewing experience.

tdabasinskas committed:
Merge branch 'main' into 1005

tdabasinskas committed:
feat(feed): add docker container metrics and update article dates

- Added new article about docker container metrics
- Updated publication dates for several articles
- Reorganized feed order to maintain chronological sorting

tdabasinskas committed:
docs(metrics-management): update OTEL collector exporters configuration

- Add highlighted syntax for cloud configuration
- Add missing self-host configuration example
- Improve placeholder variables formatting with curly braces
- Add clear instructions for replacing placeholder values

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+    tls:
+      insecure: false
+    headers:
+      signoz-access-token: "{signoz-token}"
```
```suggestion
      signoz-ingestion-key: "{signoz-ingestion-key}"
```

This was recently changed.

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
````diff
@@ -0,0 +1,199 @@
+      signoz-access-token: "{signoz-token}"
+```
+
+You would need to replace `{region}` and `{signoz-token}` in the above file with the region of your choice and token obtained from Signoz Cloud → Settings → Ingestion Settings.
````
```suggestion
You would need to replace `{region}` and `{signoz-token}` in the above file with the [region](https://signoz.io/docs/ingestion/signoz-cloud/overview/) of your SigNoz Cloud environment and [Ingestion Key ](https://signoz.io/docs/ingestion/signoz-cloud/keys/)
```

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+
+## Pre-requisites
+
+- [OTel Collector](/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/).
```
This should be the first step of the document instead of a pre-requisite.

You can remove it from here and create a section just after Pre-requisites

## Setup OTel Collector

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+    <figcaption><i>Block IO Panel</i></figcaption>
+</figure>
+
+## Conclusion
```
```suggestion

```
No need for a conclusion section here.
If there are any troubleshooting steps - general or specific that can be helpful for a user, let's add troubleshooting steps. If not, we can let it be.

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+
+## Conclusion
+
+Following this guide, you should have a working setup for collecting Docker container metrics using OTel Collector and visualizing them in SigNoz.
```
```suggestion

```

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+
+Following this guide, you should have a working setup for collecting Docker container metrics using OTel Collector and visualizing them in SigNoz.
+
+You can find the full list of metrics collected by the `docker_stats` receiver and available for querying in dashboards [in the official receiver documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/documentation.md).
```
Add this as the final sentence after Disk Metrics under the Visualizing Docker Container Metrics section.

Calm-Rock replied:
Add this as the final sentence after Disk Metrics under the Visualizing Docker Container Metrics section.

Calm-Rock commented on data/docs/metrics-management/docker-container-metrics.mdx:
```diff
@@ -0,0 +1,199 @@
+- [OTel Collector](/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/).
+- Docker API version 1.22+
+- Linux (`docker_stats` receiver does not work on Windows or Mac).
+- Either a [SigNoz Cloud account](/docs/cloud/) or a [Self-Hosted SigNoz instance](/docs/install/docker/).
```
```suggestion
```

tdabasinskas committed:
Update data/docs/metrics-management/docker-container-metrics.mdx

Co-authored-by: CheetoDa <31571545+Calm-Rock@users.noreply.github.com>

tdabasinskas committed:
Apply suggestions from code review

Co-authored-by: CheetoDa <31571545+Calm-Rock@users.noreply.github.com>

tdabasinskas committed:
chore(git): update branch

tdabasinskas committed:
docs(metrics-management): remove extra blank lines in docker metrics doc

tdabasinskas committed:
docs(metrics-management): add OTel collector setup section for Docker metrics

Added a new section explaining the prerequisite to setup OpenTelemetry Collector for Docker container metrics monitoring. This helps users understand the complete setup process by providing a link to the detailed OTel Collector tutorial.

tdabasinskas committed:
build(deps): remove explicit yarn package manager version

Removed the packageManager field from package.json as it's not necessary to lock to a specific yarn version in this context
